attendance/views.py

```diff
@@ -240,6 +240,7 @@ def fetchLatestAttendances(current_class):
     response = {}
     response["current_class"] = details
     response["all_attendance"] = json_attendance
+    response["can_mark_attendance"] = can_mark_attendance(request).content.decode("utf-8") == "true"
```

Comment:
can_mark_attendance is true for all the staff members so you can check if `request.user.is_staff`.

Also this permission is user specific and the response of this function is cached by the calling function, remove this line from here and do permission check in calling funciton. Also make sure the this field is never cached and is set for each request user.

Reply:
Once Review it @Diwakar-Gupta sir